javascripts/tagmanager.js

```diff
@@ -1830,6 +1830,20 @@
                 }
             }
 
+            // Only sync the dataLayer changes from GTM if the config has been set
+            if (!('undefined' === typeof activelySyncGtmDataLayer) && activelySyncGtmDataLayer && 'undefined' !== typeof windowAlias.dataLayer && utils.isArray(windowAlias.dataLayer)) {
```

Comment:
@snake14 this check needs to be updated, otherwise it won't work if someone tries to initialise dataLayer after the script is loaded

Case when adding dataLayer via console
![Screenshot from 2024-09-12 11-46-48](https://github.com/user-attachments/assets/b6d33841-9850-4c8d-b77d-1b8fd1a40465)

With suggested change
![Screenshot from 2024-09-12 11-48-25](https://github.com/user-attachments/assets/a79c2e30-aa4c-4a4d-8e43-568f96dc05d1)


```suggestion
            if (!('undefined' === typeof activelySyncGtmDataLayer) && activelySyncGtmDataLayer) {
                windowAlias.dataLayer = windowAlias.dataLayer || [];
```

Reply:
Thank you for the good suggestion. I have applied the change.